src/python/ib1/openenergy/support/ckan.py

```diff
@@ -188,18 +188,29 @@ def map_date(item):
 
                 # Check whether the data package already exists in CKAN
                 try:
-                    ckan.action.package_show(id=dataset_id)
+                    existing_package = ckan.action.package_show(id=dataset_id)
                     LOG.info(f'data package id={dataset_id} already exists in CKAN, updating')
                     # Found existing package, use package update
                     result = ckan.action.package_update(name=dataset_id,
                                                         **ckan_dict_from_metadata(
                                                             data_set,
                                                             description_template=description_template),
-                                                        resources=[],
                                                         owner_org=org.organisation_id,
                                                         return_package_dict=True)
+                    existing_resources = { res['name']: res for res in existing_package['resources'] }
                     for resource in resources:
-                        ckan.action.resource_create(**resource)
+                        res_name = resource['name']
+                        if res_name in existing_resources:
+                            res_id = existing_resources[res_name]['id']
+                            # package_update() above removes any existing resources,
+                            # so create a new resource with the previously existing ID.
+                            ckan.action.resource_create(id=res_id, **resource)
+                            existing_resources.pop(res_name)
+                        else:
+                            ckan.action.resource_create(**resource)
+                    for obsolete_resource_name in existing_resources:
```

Comment:
Is this step necessary if package_update removed existing resources? Or are the previous resources left orphaned?

Reply:
Good question.  With the current code, I suspect this does nothing, but I haven't checked in detail.  I originally wrote this when I was using `package_patch()` instead of `package_update()`.  I think the best thing to do is for me to write a comment above this code mentioning that it probably does nothing currently (but we should check for orphaned  resources), but if/when we move to `package_patch()` we will need it.